ingestion/ingest-google-docs.py

```diff
@@ -0,0 +1,134 @@
+    if not text_run:
+        return ''
+    return text_run.get('content')
+
+
+def read_strucutural_elements(elements):
+    """Recurses through a list of Structural Elements to read a document's text where text may be
+        in nested elements.
+
+        Args:
+            elements: a list of Structural Elements.
+    """
+    text = ''
+    for value in elements:
+        if 'paragraph' in value:
+            elements = value.get('paragraph').get('elements')
+            for elem in elements:
+                text += read_paragraph_element(elem)
+        elif 'table' in value:
+            # The text in table cells are in nested Structural Elements and tables may be
+            # nested.
+            table = value.get('table')
+            for row in table.get('tableRows'):
+                cells = row.get('tableCells')
+                for cell in cells:
+                    text += read_strucutural_elements(cell.get('content'))
+        elif 'tableOfContents' in value:
+            # The text in the TOC is also in a Structural Element.
+            toc = value.get('tableOfContents')
+            text += read_strucutural_elements(toc.get('content'))
+    return text
+
+
+def run():
+    # Find ids of all Google docs in the raw Serge folder
+    store = file.Storage('secrets/token_read.json')
+    page_token = None   
+    creds = None
+    flow = client.flow_from_clientsecrets('secrets/credentials.json', SCOPES)
+    if os.path.exists('secrets/token_read.pickle'):
+        with open('secrets/token_read.pickle', 'rb') as token:
+            creds = pickle.load(token)
+        # If there are no (valid) credentials available, let the user log in.
+    if not creds or creds.invalid:
+        if creds and creds.expired and creds.refresh_token:
+            creds.refresh(Request())
+        else:
+            flow = client.flow_from_clientsecrets('secrets/credentials.json', SCOPES)
+            creds = tools.run_flow(flow, store)
+        # Save the credentials for the next run
+        with open('secrets/token_read.pickle', 'wb') as token:
+            pickle.dump(creds, token)
+    service_drive = build('drive', 'v3', credentials=creds)
+    file_ids = []
+    while True:
+        response = service_drive.files().list(q="name contains 'Raw_Document'",
+                                              spaces='drive',
+                                              fields='nextPageToken, files(id, name)',
+                                              pageToken=page_token).execute()
+        for file_content in response.get('files', []):
+            # Process change
+            file_id = file_content.get('id')
+            print('Found file: %s (%s)' % (file_content.get('name'), file_id))
+            file_ids.append(file_id)
+        page_token = response.get('nextPageToken', None)
+        if page_token is None:
+            break
+
+    # Enumerate over Google doc ids and fetch content
+    store = file.Storage('secrets/token.json')

```

Comment:
Declare new variable store_doc.

Could token.json be named token_write.json?

Reply:
Yep, cleared up the naming for all this